helix-core/src/test/java/org/apache/helix/integration/rebalancer/TestOnDemandRebalancer.java

```diff
@@ -0,0 +1,81 @@
+package org.apache.helix.integration.rebalancer;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import org.apache.helix.AccessOption;
+import org.apache.helix.HelixDataAccessor;
+import org.apache.helix.PropertyKey;
+import org.apache.helix.integration.task.MockTask;
+import org.apache.helix.integration.task.TaskTestBase;
+import org.apache.helix.integration.task.WorkflowGenerator;
+import org.apache.helix.task.JobConfig;
+import org.apache.helix.task.TaskConstants;
+import org.apache.helix.task.TaskState;
+import org.apache.helix.task.Workflow;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * This test check the funcionality of OnDemandRebalancer
+ */
+public class TestOnDemandRebalancer extends TaskTestBase {
+  @Test
+  public void testOndemandRebalancer() throws Exception {
```

Comment:
@alirezazamani 

How could this be testing the RebalanceUtil's rebalance functionality when it never calls it in the test? Please write a test that calls RebalanceUtil explicitly to ensure that the controller indeed triggers rebalance.

You should also try to mix & match all possible input combinations to the function you're trying to test and make sure it's behaving correctly.

As I reiterated multiple times, let's please try to think about the real purpose of having test coverage instead of blindly copying & pasting.

Reply:
@alirezazamani 

Another point: you might find the Mockito framework useful when mocking cluster components. https://www.vogella.com/tutorials/Mockito/article.html